### PR TITLE
test: add tests that passes data through Plug.Parsers

### PR DESCRIPTION
**Pull request by hauleth**
This also adds tests for `identity` content encoding and for situation
when there is no content encoding header at all.

### CODE REVIEW

File: lib/plug_caisson.ex [hunk 1]
```diff
@@ -13,7 +13,9 @@ defmodule PlugCaisson do
   @callback init(opts :: term()) :: {:ok, state :: term()} | {:error, term()}
   @callback deinit(state :: term()) :: term()
   @callback process(state :: term(), data :: binary(), opts :: keyword()) ::
-              {:ok, binary()} | {:more, binary()} | {:error, term()}
+              {:ok, binary(), new_state :: term()}
```
Comment (Ziinc) [inline lib/plug_caisson.ex]: Should document how state can be used by custom algos, and how it can be utilized in downstream plugs in the pipeline.

**Review by Ziinc — CHANGES_REQUESTED** [review]
adding testing and the private state management seem like separate concerns, should separate out to different PRs.
Also, missing testing for custom state setting.

File: lib/plug_caisson.ex (hunk 1 above)
Comment (hauleth) [reply lib/plug_caisson.ex]: I will move that to separate PR.

File: lib/plug_caisson.ex [hunk 2]
```diff
@@ -89,6 +89,9 @@ defmodule PlugCaisson do
           _ ->
```
Comment (Ziinc) [inline lib/plug_caisson.ex:89]: small nit: this entire `fetch_decompressor/2` block should be refactored to a `with ... do...else` with the `content_encoding` case as the happy path and all other situations under `else`. The 3 level nesting affects readability quite a bit.

**Review by Ziinc — APPROVED** [review]
looks great. Can consolidate the repetitive tests in a follow up PR, this looks good for announcement now.

**Pull request merged**